In [1]:
import pandas as pd
df = pd.read_csv('OnlineRetail.csv', encoding = "ISO-8859-1")
df.head()

InvoiceNo StockCode                          Description  Quantity  \
0    536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1    536365     71053                  WHITE METAL LANTERN         6   
2    536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3    536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4    536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   

      InvoiceDate  UnitPrice  CustomerID         Country  
0  12/1/2010 8:26       2.55     17850.0  United Kingdom  
1  12/1/2010 8:26       3.39     17850.0  United Kingdom  
2  12/1/2010 8:26       2.75     17850.0  United Kingdom  
3  12/1/2010 8:26       3.39     17850.0  United Kingdom  
4  12/1/2010 8:26       3.39     17850.0  United Kingdom

### 1. Đọc bộ dữ liệu, cho biết số dòng, số cột và kiểu dữ liệu của các thuộc tính

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    541909 non-null  object 
 1   StockCode    541909 non-null  object 
 2   Description  540455 non-null  object 
 3   Quantity     541909 non-null  int64  
 4   InvoiceDate  541909 non-null  object 
 5   UnitPrice    541909 non-null  float64
 6   CustomerID   406829 non-null  float64
 7   Country      541909 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 33.1+ MB


+ Số dòng: 541909 dòng, 8 cột
+ InvoiceNo, StockCode, Description, InvoiceDate, Country, CustomerID: dữ liệu định tính
+ Quantity, UnitPrice: dữ liệu định lượng, thang đo tỷ lệ

### 2. Trích xuất dữ liệu các cột Description và Quantity lưu vào file OnlineRetail.csv

In [5]:
tg = df.iloc[:,2:4]
tg.to_csv('OnelineRetail2.csv')

### 3. Trích xuất dữ liệu 1000 dòng đầu tiên lưu vào file OnlineRetail.xlsx

In [6]:
tg = df.iloc[:1000]
tg.to_csv('OnelineRetail2.xlsx')

### 4. Trích xuất các bản ghi có số lượng từ 10 trở lên lưu vào file OnlineRetail.h5

In [14]:
tg = df.loc[df.Quantity >=10]
tg

InvoiceNo StockCode                      Description  Quantity  \
9         536367     84879    ASSORTED COLOUR BIRD ORNAMENT        32   
26        536370     22728        ALARM CLOCK BAKELIKE PINK        24   
27        536370     22727        ALARM CLOCK BAKELIKE RED         24   
28        536370     22726       ALARM CLOCK BAKELIKE GREEN        12   
29        536370     21724  PANDA AND BUNNIES STICKER SHEET        12   
...          ...       ...                              ...       ...   
541894    581587     22631         CIRCUS PARADE LUNCH BOX         12   
541895    581587     22556   PLASTERS IN TIN CIRCUS PARADE         12   
541896    581587     22555        PLASTERS IN TIN STRONGMAN        12   
541902    581587     22629              SPACEBOY LUNCH BOX         12   
541904    581587     22613      PACK OF 20 SPACEBOY NAPKINS        12   

            InvoiceDate  UnitPrice  CustomerID         Country  
9        12/1/2010 8:34       1.69     13047.0  United Kingdom  
26       12/1/2010 8:45       3.75     12583.0          France  
27       12/1/2010 8:45       3.75     12583.0          France  
28       12/1/2010 8:45       3.75     12583.0          France  
29       12/1/2010 8:45       0.85     12583.0          France  
...                 ...        ...         ...             ...  
541894  12/9/2011 12:50       1.95     12680.0          France  
541895  12/9/2011 12:50       1.65     12680.0          France  
541896  12/9/2011 12:50       1.65     12680.0          France  
541902  12/9/2011 12:50       1.95     12680.0          France  
541904  12/9/2011 12:50       0.85     12680.0          France  

[154919 rows x 8 columns]

### 5. Trích xuất dữ liệu từ các phẩn tử từ dòng 1000 đến dòng 2000, các cột Quantity, InvoiceDate, UnitPrice lưu vào file OnlineRetail.json

In [12]:
tg = df.iloc[:,3:6][1000:2000]
tg.to_csv('OnelineRetail.json')

### 6. Trích xuất các bản ghi có số hoá đơn '536365', lưu vào file OnlinerRetail.html

In [9]:
tg = df.loc[df.InvoiceNo == '536365']
tg.to_csv('OnlineRetail.html')